packages/gatsby/src/utils/restarting-screen.ts

```diff
@@ -96,12 +96,22 @@ export default html`
                   ":" +
                   services.developstatusserver.port
               )
-              socket.on("develop:started", () => {
-                window.location.reload()
-              })
+              socket.on("structured-log", msg => {
+                if (msg.type !== "LOG_ACTION") return
+
+                if (
+                  msg.action.type === "SET_STATUS" &&
+                  msg.action.payload === "SUCCESS"
```

Comment:
What should happen if it's not success? (it can be `FAILED` for multiple reasons - this could some failure when doing webpack compilation, or during query extraction).

Also - we emit this potentially multiple times after starting `gatsby develop`, so this  could mean that each webpack compilation (hot reloads) actually force hard reload I think

Reply:
We should add an error state to this screen, good call! Let's do this after this PR as that's the current behavior as well and I don't want this branch to be forever long lived :grimacing:

Reply:
#26064